### PR TITLE
[Enhancement] Support Hive OpenCSVSerde

### PR DESCRIPTION
**Pull request by Smith-Cruise**
If the user create a hive CSV table with the below grammar:
```sql
CREATE EXTERNAL TABLE `csv_wrong_parse`(
  `open` string COMMENT 'from deserializer',
  `union` string COMMENT 'from deserializer',
  `id` string COMMENT 'from deserializer')
COMMENT '数据'
ROW FORMAT SERDE
  'org.apache.hadoop.hive.serde2.OpenCSVSerde'
WITH SERDEPROPERTIES (
  'escapeChar'='\\',
  'quoteChar'='"',
  'separatorChar'=',',
  'serialization.encoding'='GBK')
STORED AS INPUTFORMAT
  'org.apache.hadoop.mapred.TextInputFormat'
OUTPUTFORMAT
  'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat';
```

SR will not recognize `separatorChar` parameter as a field delimiter.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by dirtysalt — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java [hunk 1]
```diff
@@ -333,14 +333,19 @@ public static TextFileFormatDesc toTextFileFormatDesc(Map<String, String> serdeP
         // https://issues.apache.org/jira/browse/HIVE-16922
         String collectionDelim;
         if (serdeParams.containsKey("colelction.delim")) {
-            collectionDelim = serdeParams.get("colelction.delim");
+            collectionDelim = serdeParams.getOrDefault("colelction.delim", "");
```
Comment (letian-jiang) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java:336]: typo?

**Review by letian-jiang — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|6|7|85.71%|[336]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-2.5

**Pull request merged**